apps/ledger-live-desktop/src/renderer/screens/stake/index.tsx

```diff
@@ -1,45 +1,19 @@
-import { useState, useEffect, useMemo } from "react";
 import SelectAccountAndCurrencyDrawer from "~/renderer/drawers/DataSelector/SelectAccountAndCurrencyDrawer";
 import { setDrawer } from "~/renderer/drawers/Provider";
 import { useFeature } from "@ledgerhq/live-common/featureFlags/index";
 import { Account } from "@ledgerhq/types-live";
-import perFamilyManageActions from "~/renderer/generated/AccountHeaderManageActions";
 import { useHistory } from "react-router-dom";
 import { stakeDefaultTrack } from "./constants";
 import { track, page } from "~/renderer/analytics/segment";
+import { useDispatch } from "react-redux";
+import { openModal } from "~/renderer/actions/modals";
 
-type Props = {
-  account?: Account;
-  parentAccount?: Account | null;
-};
-
-type CurrencyTypes = keyof typeof perFamilyManageActions;
-
-const useStakeFlow = (props: Props) => {
+const useStakeFlow = () => {
   const history = useHistory();
-  const [accountSelection, setAccountSelection] = useState<Props>({});
   const stakeProgramsFeatureFlag = useFeature("stakePrograms");
-  const { params: paramsFlag, enabled: stakeProgramsEnabled } = stakeProgramsFeatureFlag || {};
+  const dispatch = useDispatch();
+  const { params: paramsFlag } = stakeProgramsFeatureFlag || {};
   const { list: listFlag } = paramsFlag || {};
```

Comment:
```suggestion
    const { params: paramsFlag, enabled: stakeProgramsEnabled } = stakeProgramsFeatureFlag || {};
  const { list: listFlag } = (stakeProgramsEnabled && paramsFlag) || {};
```

Reply:
Not sure why we want to do this? This hook now just provides a callback for other buttons to use, and they could always be disabled by this `enabled` field if required.

Reply:
I am not sure if set a list if the flag is disabled. 
An error in the code could be mixed with an error in the flag configuration.

Reply:
If the flag is disabled, I think the stake buttons should be disabled, otherwise we will open a drawer and display zero currencies.